src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+    def row_margin(self):
+        return self._calculator.row_margin
+
+    @lazyproperty
+    def column_margin(self):
+        return self._calculator.column_margin
+
+    @lazyproperty
+    def table_margin(self):
+        return self._calculator.table_margin
+
+    @lazyproperty
+    def row_base(self):
+        return self._calculator.row_base
+
+    @lazyproperty
+    def column_base(self):
+        return self._calculator.column_base
+
+    @lazyproperty
+    def table_base(self):
+        return self._calculator.table_base
+
+    @lazyproperty
+    def table_percentages(self):
+        return self.table_proportions * 100
+
+    @lazyproperty
+    def column_percentages(self):
+        return self.column_proportions * 100
+
+    @lazyproperty
+    def row_percentages(self):
+        return self.row_proportions * 100
+
+    @lazyproperty
+    def population_counts(self):
+        return (
+            self.table_proportions * self._population * self._cube.population_fraction
+        )
+
+    @lazyproperty
+    def row_labels(self):
+        return self._calculator.row_labels
+
+    @lazyproperty
+    def column_labels(self):
+        return self._calculator.column_labels
+
+    # Properties ---------------------------------------------------------------------
+
+    @lazyproperty
+    def _ordering(self):
+        return (
+            OrderTransform(self._cube.dimensions, self._reordered_ids)
+            if self._reordered_ids
+            else None
+        )
+
+    @lazyproperty
+    def _transforms(self):
+        return Transforms(self._ordering, self._pruning, self._insertions)
+
+    @lazyproperty
+    def _assembler(self):
+        return Assembler(self._slice, self._transforms)
+
+    @lazyproperty
+    def _calculator(self):
+        return Calculator(self._assembler)
+
+    @lazyproperty
+    def _dimensions(self):
+        dimensions = self._cube.dimensions[-2:]
+        if self._ca_as_0th:
+            return tuple([dimensions[-1]])
+        return dimensions
+
+    @lazyproperty
+    def _insertions(self):
+        return (
+            Insertions(self._dimensions, self._slice) if self._use_insertions else None
+        )
+
+    @lazyproperty
+    def _slice(self):
+        dimensions = self._cube.dimensions[-2:]
+        base_counts = self._cube._apply_missings(
+            self._cube._measure(False).raw_cube_array
+        )
+        counts = self._cube._apply_missings(
+            self._cube._measure(self._weighted).raw_cube_array
+        )
+        type_ = self._cube.dim_types[-2:]
+        if self._cube.ndim > 2 or self._ca_as_0th:
+            base_counts = base_counts[self._slice_idx]
+            counts = counts[self._slice_idx]
+            if self._cube.dim_types[0] == DT.MR:
+                base_counts = base_counts[0]
+                counts = counts[0]
+            elif self._ca_as_0th:
+                return _1DCatSlice(self._dimensions, counts, base_counts)
+        elif self._cube.ndim < 2:
```

Comment:
We have essentially 2 types of 1D stands, a Cat one and an MR one. CA as 0th is essentially a 1D Cat.